server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -2231,57 +2231,34 @@ private void deleteSnapshotsFromRepository(
-            threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(new AbstractRunnable() {
-                @Override
-                public void onFailure(Exception e) {
-                    checkForShallowSnapshotStep.onFailure(e);
-                }
-
-                @Override
-                protected void doRun() {
-                    boolean cleanupRemoteStoreLockFiles = false;
-                    final List<SnapshotId> allSnapshotIds = repositoryData.getSnapshotIds().stream().collect(Collectors.toList());
-                    for (SnapshotId snapshotId : allSnapshotIds) {
-                        if (Boolean.TRUE.equals(repository.getSnapshotInfo(snapshotId).isRemoteStoreIndexShallowCopyEnabled())) {
-                            cleanupRemoteStoreLockFiles = true;
-                            break;
-                        }
-                    }
-                    checkForShallowSnapshotStep.onResponse(cleanupRemoteStoreLockFiles);
-                }
-            });
 
-            checkForShallowSnapshotStep.whenComplete(cleanupRemoteStoreLockFiles -> {
-                if (cleanupRemoteStoreLockFiles) {
-                    repository.deleteSnapshotsAndReleaseLockFiles(
-                        snapshotIds,
-                        repositoryData.getGenId(),
-                        minCompatibleVersion(minNodeVersion, repositoryData, snapshotIds),
-                        remoteStoreLockManagerFactory,
-                        ActionListener.wrap(updatedRepoData -> {
-                            logger.info("snapshots {} deleted", snapshotIds);
-                            removeSnapshotDeletionFromClusterState(deleteEntry, null, updatedRepoData);
-                        }, ex -> removeSnapshotDeletionFromClusterState(deleteEntry, ex, repositoryData))
-                    );
-                } else {
-                    repository.deleteSnapshots(
-                        snapshotIds,
-                        repositoryData.getGenId(),
-                        minCompatibleVersion(minNodeVersion, repositoryData, snapshotIds),
-                        ActionListener.wrap(updatedRepoData -> {
-                            logger.info("snapshots {} deleted", snapshotIds);
-                            removeSnapshotDeletionFromClusterState(deleteEntry, null, updatedRepoData);
-                        }, ex -> removeSnapshotDeletionFromClusterState(deleteEntry, ex, repositoryData))
-                    );
-                }
-            }, ex -> removeSnapshotDeletionFromClusterState(deleteEntry, ex, repositoryData));
+            // TODO: Relying on repository flag to decide delete flow may lead to shallow snapshot blobs not being taken up for cleanup
+            // when the repository currently have the flag disabled and we try to delete the shallow snapshots taken prior to disabling
+            // the flag. This can be improved by having the info whether there ever were any shallow snapshot present in this repository
```

Comment:
can you link the github issue as well tracking this change ?

Reply:
Added in the latest revision